### DISCUSSION TITLE
Convert to NNEF

### DISCUSSION THREAD

**Discussion by igor-yusupov** [sonos/tract#1320] [Q&A] [answered]
I want to convert to NNEF format weights to have faster loading with command: `tract encoder.onnx -i src:sequence,1,f32 -i src_mask:sequence,sequence,bool dump --nnef encoder.nnef.tgz`

But I got error:
```
Caused by:
        0: Translating model to proto_model
        1: Translating model to AST
        2: No serializer found for node #0 "src" Source
``` 

Can you help me to solve the problem please? I can attach weights if needed

**Comment by kali** [comment, marked as answer]
You need to add `--nnef-tract-core`. This will enable tract extensions. Your source node (tract representation of a model input) is non-NNEF as it features symbolic dimensions.

**Reply by igor-yusupov** [reply]
Thank you, I succeed in converting model. But now I got error: `Registry not found Identifier("tract_core")`

code of reading model:

```
let encoder = tract_nnef::nnef()
        .model_for_path("encoder.nnef.tgz")
        .unwrap()
        // .with_output_fact(0, InferenceFact::default())
        // .unwrap()
        .into_optimized()
        .unwrap()
        .into_runnable()
        .unwrap();
```

**Comment by kali** [comment]
Same thing, really. Now your model uses tract extension, you need to add `.with_tract_core()` after the `nnef()` instantiation to enable them.

**Reply by igor-yusupov** [reply]
I did it. Thank you very much!
